src/besselj.jl

```diff
@@ -238,3 +264,71 @@ function besselj_debye(v, x)
 
     return coef * Uk_poly_Jn(p, v, p2, T)
 end
+
+# For 0.0 <= x < 171.5
+# Mean ULP = 0.55
+# Max ULP = 2.4
+# Adapted from Cephes Mathematical Library (MIT license https://en.smath.com/view/CephesMathLibrary/license) by Stephen L. Moshier
+function gamma(x)
+    if x > 11.5
+        return large_gamma(x)
+    elseif x < 0.0
+        #p = floor(x)
+        #isequal(p, abs(x)) && return throw(DomainError(x, "NaN result for non-NaN input."))
+        # need reflection formula
+        return throw(DomainError(x, "Negative numbers are currently not implemented"))
+    elseif x <= 11.5
+        return small_gamma(x)
+    elseif isnan(x)
+        return x
+    end
+end
+function large_gamma(x)
+    T = Float64
+    w = inv(x)
+    s = (
+        8.333333333333331800504e-2, 3.472222222230075327854e-3, -2.681327161876304418288e-3, -2.294719747873185405699e-4,
+        7.840334842744753003862e-4, 6.989332260623193171870e-5, -5.950237554056330156018e-4, -2.363848809501759061727e-5,
+        7.147391378143610789273e-4
+    )
+    w = w * evalpoly(w, s) + one(T)
+    # lose precision on following block
+    y = exp((x)) 
+    if x > 143.01608
+        isinf(x) && return x
+        # avoid overflow
+        v = x^(0.5 * x - 0.25)
```

Comment:
I think that just doing this unconditionally is likely to be better for performance.

Reply:
Sounds good! We lose some precision here so we might be worth doing a routine like yours above in the thread. Ideally, we could have a version that would work on the long term stable release.